File: December - 05/KishoreMuruganantham_island_counter.swift
import Foundation

var lines = [String]()
while let line = readLine() { lines.append(line) }

if !lines.isEmpty {
    let dims = lines[0].split(separator: " ").compactMap { Int($0) }
    let r = dims[0]
    let c = dims[1]
    
    var grid = [[Int]]()
    for i in 1...r {
        let row = lines[i].split(separator: " ").compactMap { Int($0) }
        grid.append(row)
    }
    
    var visited = Array(repeating: Array(repeating: false, count: c), count: r)
    var count = 0
    let dr = [0, 0, 1, -1]
    let dc = [1, -1, 0, 0]
    
    for i in 0..<r {
        for j in 0..<c {
            if grid[i][j] == 1 && !visited[i][j] {
                count += 1
                var q = [(i, j)]
                visited[i][j] = true
                var head = 0
                while head < q.count {
                    let (cr, cc) = q[head]
                    head += 1
                    
                    for k in 0..<4 {
                        let nr = cr + dr[k]
                        let nc = cc + dc[k]
                        
                        if nr >= 0 && nr < r && nc >= 0 && nc < c {
                            if grid[nr][nc] == 1 && !visited[nr][nc] {
                                visited[nr][nc] = true
                                q.append((nr, nc))
                            }
                        }
                    }
                }
            }
        }
    }
    print(count)
}

File: December - 05/KishoreMuruganantham_island_counter.R
input <- file("stdin", "r")
lines <- readLines(input)
dims <- as.integer(unlist(strsplit(lines[1], "\\s+")))
r <- dims[1]
c <- dims[2]

grid <- matrix(0, nrow = r, ncol = c)
for (i in 1:r) {
  row_vals <- as.integer(unlist(strsplit(lines[i + 1], "\\s+")))
  grid[i, ] <- row_vals
}

visited <- matrix(FALSE, nrow = r, ncol = c)
icount <- 0

dr <- c(0, 0, 1, -1)
dc <- c(1, -1, 0, 0)

q_r <- integer(r * c + 1)
q_c <- integer(r * c + 1)

for (i in 1:r) {
  for (j in 1:c) {
    if (grid[i, j] == 1 && !visited[i, j]) {
      icount <- icount + 1
      
      head <- 1
      tail <- 1
      
      q_r[tail] <- i
      q_c[tail] <- j
      tail <- tail + 1
      visited[i, j] <- TRUE
      
      while (head < tail) {
        cr <- q_r[head]
        cc <- q_c[head]
        head <- head + 1
        
        for (k in 1:4) {
          nr <- cr + dr[k]
          nc <- cc + dc[k]
          
          if (nr >= 1 && nr <= r && nc >= 1 && nc <= c) {
            if (grid[nr, nc] == 1 && !visited[nr, nc]) {
              visited[nr, nc] <- TRUE
              q_r[tail] <- nr
              q_c[tail] <- nc
              tail <- tail + 1
            }
          }
        }
      }
    }
  }
}

cat(icount, "\n", sep = "")
